pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
Agree that it makes sense to make it extensible.
Does something like this work?
```go

type struct InterfaceInfo {
    Iface net.Interface
    IPAddrs []net.IP
}

type struct NetworkStatus {
    IPv4 net.IP  // Primary ipv4 address of the pod
    IPv6 net.IP  // Primary ipv6 address of the pod
    NetworkInterfaces []net.InterfaceInfo
}
```